src/main/scala/is/hail/variant/VariantView.scala

```diff
@@ -0,0 +1,56 @@
+package is.hail.variant
+
+import is.hail.annotations._
+import is.hail.expr._
+
+class VariantView(t: TStruct) {
```

Comment:
Shouldn't this take a `TVariant`?  You don't need to be decomposing arbitrary structs as `Variant`s, just `TVariant`'s representation.